src/DataStructures/Tensor/Expressions/Product.hpp

```diff
@@ -233,3 +234,41 @@ SPECTRE_ALWAYS_INLINE auto operator*(
   return TensorExpressions::contract(
       TensorExpressions::OuterProduct<T1, T2>(~t1, ~t2));
 }
+
+/// \ingroup TensorExpressionsGroup
+/// \brief Returns the tensor expression representing the product of a tensor
+/// expression and a scalar
+///
+/// \tparam T the derived TensorExpression type of the tensor expression operand
+/// of the product
+/// \tparam ArgsList the TensorIndexs of the tensor expression operand
+/// \param t the tensor expression operand of the product
+/// \param s the scalar operand of the product
+/// \return the tensor expression representing the product of a tensor
+/// expression and a scalar
+template <typename T, typename ArgsList>
+SPECTRE_ALWAYS_INLINE auto operator*(
+    const TensorExpression<T, double, typename T::symmetry,
+                           typename T::index_list, ArgsList>& t,
+    const double s) {
+  return t * TensorExpressions::ScalarAsExpression(s);
```

Comment:
can you instead make this `ScalarAsExpression(s) * t`? I think we discussed having the scalars on the LHS at some point

Reply:
I considered doing this switch originally and then I didn't because - being completely honest - although I felt convinced it would matter when we briefly discussed it, in tracing through the syntax tree in my head, I didn't really see what the practical difference would be. Since I didn't see a perceived difference, I left the original order from the user intact because then the generated tree matches what they wrote and would be easier to debug.

If there would be a practical difference in permuting these operands, happy to change it! If so, could you provide a case where you feel like reordering would affect optimization/accuracy? I'm generally aware that multiplication order can affect the end result, but wasn't sure what the benefit of doing this switch would be. Let me know!